### PR TITLE
publishing collected_sources.tar.gz to eve-sources repo under docker/lfedge

### PR DESCRIPTION
**Pull request by yash-zededa**
This change will push the collected_sources artefact created for the SBOM comparison to eve-sources docker repo under lfedge. eve-sources will be a new repo and a ticket is already in place to create the new repo by @eriknordmark 

Added a new target `publish_sources` that depends on `collected_sources`.

`publish_sources` will build the docker image from pkg/sources/Dockerfile and push to lfedge/eve-sources docker hub repo

### CODE REVIEW

File: pkg/kernel/kernel-config/kernel_config-5.10.x-aarch64 [hunk 1]
```diff
@@ -6462,7 +6465,15 @@ CONFIG_STM_SOURCE_CONSOLE=y
 
 # CONFIG_FPGA is not set
 # CONFIG_FSI is not set
-# CONFIG_TEE is not set
+CONFIG_TEE=m
```
Comment (deitch) [inline pkg/kernel/kernel-config/kernel_config-5.10.x-aarch64:6468]: Not sure why this is here. We should removal unrelated changes from this PR.

File: pkg/sources/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Collected_sources.tar.gz includes:
+
+* alpine, kernel and golang packages.
+* A manifest file collected_sources_manifest.csv that holds the metadata for all the packages.
+
+## Usage
+
+```bash
+make -e V=1 HV=${{ matrix.hv }} ZARCH=${{ matrix.arch }} LINUXKIT_PKG_TARGET=push publish_sources
```
Comment (deitch) [inline pkg/sources/README.md:13]: Can you add something about how this does not run locally, but rather is copied to `SOURCES_DIR` from the `Makefile`, e.g. `dist/amd64/0.0.0-master-..../sources/` and run there, so no one gets confused?

**Review by deitch — CHANGES_REQUESTED** [review]
Looks like one accidental unrelated change, and a request to add to the README. Other than that, really looks good.

File: pkg/kernel/kernel-config/kernel_config-5.10.x-aarch64 (hunk 1 above)
Comment (yash-zededa) [reply pkg/kernel/kernel-config/kernel_config-5.10.x-aarch64:6468]: Yes, already on it.

File: pkg/sources/README.md (hunk 2 above)
Comment (yash-zededa) [reply pkg/sources/README.md:13]: Will add more info.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -121,6 +121,10 @@ jobs:
       - name: Build and push EVE for ${{ matrix.arch }}-${{ matrix.hv }}
         run: |
           make -e V=1 HV=${{ matrix.hv }} ZARCH=${{ matrix.arch }} LINUXKIT_PKG_TARGET=push eve
+      - name: Build and push collected_sources to eve-source repo
+        run: |
+          make sbom collected_sources compare_sbom_collected_sources
+          make -e V=1 HV=${{ matrix.hv }} ZARCH=${{ matrix.arch }} LINUXKIT_PKG_TARGET=push publish_sources
```
Comment (eriknordmark) [inline .github/workflows/publish.yml:127]: What docket hub path will this publish to?

Comment (deitch) [reply .github/workflows/publish.yml:127]: Relies on the [build.yml](https://github.com/lf-edge/eve/pull/3209/files#diff-6baf960b0af2df8c9669a02f76c91ed0d96a76602abd86f3b3f02b1290cad5d0R1), identical to how the `make eve` in the previous few lines of `publish.yml` work.

In this case, it is `lfedge/eve-sources`. If that should change, just comment here, and it can be updated.

File: pkg/sources/README.md (hunk 2 above)
Comment (yash-zededa) [reply pkg/sources/README.md]: added more around the workflow and process

**Review by deitch — APPROVED** [review]
LGTM. We do need to wait for the new lfedge Docker Hub repository, and ensure that it matches what is in `build.yaml` (if not, then change `build.yaml` content), but other than that, looks great!

File: pkg/sources/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# pkg/sources
+
+This package will build a docker image that will store the collected_sources.tar.gz which is needed for SBOM comparison. The docker image will be pushed to `eve-sources` repo under lfedfge on dockerhub.
+
+Collected_sources.tar.gz includes:
+
+* alpine, kernel and golang packages.
+* A manifest file collected_sources_manifest.csv that holds the metadata for all the packages.
+
+### Workflow
```
Comment (eriknordmark) [inline pkg/sources/README.md:10]: @yash-zededa can you address these markdownlint complaints?

Comment (yash-zededa) [reply pkg/sources/README.md]: @eriknordmark README.md linting is fixed.

**Review by eriknordmark — APPROVED** [review]
We have eve-sources on docker hub so proceeding. Let's see how the assets workflow works.
